### DISCUSSION TITLE
Why does "colour.sd_multi_leds" definition seem to produces wrong spectrum when using wavelengths that are very close / overlapping gaussians?

### DISCUSSION THREAD

**Discussion by physici** [colour-science/colour#1104] [General]
### Description

I measure the reflectance of objects based on multi-plexed illumination with a variety of LEDs. A couple of LEDs are rather close in their peak emission and the respective gaussians overlap. However, as the LEDs are switched on in sequence, the spectral data of the reflectance is not overlapping.

When using the sd_multi_leds  function to construct the spectral distribution, the spectrum gets distorted as all gaussians of the LEDs are summed. This results in a over-rating of the LEDs that are close together.

When looking at the attached image, the code below produces only 6 peaks where 7 should be depicted. Additionally, the peak in the blue-region is twice as high as expected. 

How would I create the correct spectral distribution?

![wrong_spectrum](https://user-images.githubusercontent.com/15417017/218686081-df71ad03-b507-42a9-8811-0fcf20098f71.png)

### Code for Reproduction

```python
import colour
WAVELENGTHS = [423, 465, 472, 500, 540, 610, 635]
FWHM = [57, 18, 20, 25, 40, 12, 16]
CMFS = colour.MSDS_CMFS["CIE 1931 2 Degree Standard Observer"]
ILLUMINANT = colour.SDS_ILLUMINANTS["D65"]

power = [1, 1, 1, 1, 1, 1, 1]

spectral_distribution = colour.sd_multi_leds(WAVELENGTHS, FWHM, power)
```

### Environment Information

```shell
===============================================================================
*                                                                             *
*   Interpreter :                                                             *
*       python : 3.10.10 (tags/v3.10.10:aad5f6a, Feb  7 2023, 17:20:36)       *
*   [MSC v.1929 64 bit (AMD64)]                                               *
*                                                                             *
*   colour-science.org :                                                      *
*       colour : 0.4.2                                                        *
*                                                                             *
*   Runtime :                                                                 *
*       imageio : 2.25.0                                                      *
*       matplotlib : 3.6.3                                                    *
*       numpy : 1.24.1                                                        *
*       pandas : 1.5.3                                                        *
*       scipy : 1.10.0                                                        *
*                                                                             *
===============================================================================
```

**Comment by thomasmansencal** [comment]
Hi @physici,

The "blending-like" effect that you see is expected and happens in real-life also. Here are for examples measurements of a LED Wall primaries and its white, not how the individual primaries do not overlap as much as their combined output:

![image](https://user-images.githubusercontent.com/99779/218836188-25b1f139-8b2e-4664-bc50-df980892d6fa.png)

If one was to sum the irradiance of the individual primaries and compare it to the measured white, he would see that it is exactly, disregarding measurement variance, the same as the measured white:

![image](https://user-images.githubusercontent.com/99779/218836584-7ae401f6-1229-4225-8821-15a887eb5e42.png)

> NOTE: The second "red" entry in the legend is incorrectly named and is actually the sum of the individual primaries.

Cheers,

Thomas

**Reply by physici** [reply]
Hello Thomas,
Thanks for the explanation. I agree with you 100% as long as you are talking about emission spectra. However, I have a reflection spectrum. It was generated by illumination with LEDs on after another. In that case the "powers" do not add up. But I think it is my fault in not understanding the concept of this multi-LED spectral distribution function.

I have switched to building the spectrum from a dictionary using the colour.SpectralDistribution-function and led it interpolate the missing values. It works reasonable well and results in the correct color coordinates.

Cheers,
Rainer

**Reply by thomasmansencal** [reply]
Hello,

Right, yes this function is meant to model LED irradiance. What is not clear to me in your description is the fact that you measure reflectance with LEDs: Are you measuring the spectral reflectance for each LED primary or are you obtaining a triplet with a colorimeter and then using them to reconstruct the spectra?

Cheers,

Thomas

**Reply by physici** [reply]
I have a set of LEDs with wavelengths between 420nm and 630nm. With these LEDs I illuminate my object and measure the amount of light reflected at the surface with a photodiode. On LED is switched on at a time. I reference this data to the data captured on a white reflectance standard and calculate the color coordinates from this reflectance spectrum. My goal is to build a poor-mens-colorphotometer.
